### PR TITLE
fix strange behavior in SA1120 code fix

### LINKED ISSUE #1170: Strange behavior for SA1120 code fix within method
## Input

To reproduce this issue, run a "fix all in current document" operation on the following code:

```csharp
        private IEnumerable<int> Build(int argument)
        {
            //
            // Text line 1
            // Text line 2
            //
            return Enumerable.Range(1, 10);
        }
```

## Expected result

```csharp
        private IEnumerable<int> Build(int argument)
        {
            // Text line 1
            // Text line 2
            return Enumerable.Range(1, 10);
        }
```

## Actual result

```csharp
        private IEnumerable<int> Build(int argument)
        {
                        // Text line 1
            // Text line 2
                        return Enumerable.Range(1, 10);
        }
```

### PR DESCRIPTION
**Pull request by Noryoko**
- [x] Add regression tests
- [x] Fixes #1170 and #1171

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1120CodeFixProvider.cs [hunk 1]
```diff
@@ -113,12 +96,12 @@ private static bool TriviaHasLeadingContentOnLine(SyntaxNode root, IReadOnlyList
             return false;
```
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1120CodeFixProvider.cs]: The if and return statements above can be replaced by a single return statement.

File: StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1120CodeFixProvider.cs [hunk 2]
```diff
@@ -113,12 +96,12 @@ private static bool TriviaHasLeadingContentOnLine(SyntaxNode root, IReadOnlyList
...
 
-        private static bool TriviaHasTrailingContentOnLine(SyntaxNode root, IReadOnlyList<SyntaxTrivia> triviaList)
+        private static bool TriviaHasTrailingContentOnLine(SyntaxNode root, SyntaxTrivia commentTrivia)
         {
-            var nodeAfterTriviaStart = triviaList[triviaList.Count - 1].SpanStart - 1;
+            var nodeAfterTriviaStart = commentTrivia.Span.End + 1;
             var nodeAfterTrivia = root.FindNode(new Microsoft.CodeAnalysis.Text.TextSpan(nodeAfterTriviaStart, 1));
 
-            if (nodeAfterTrivia.GetLineSpan().StartLinePosition.Line == triviaList[triviaList.Count - 1].GetLineSpan().EndLinePosition.Line)
+            if (nodeAfterTrivia.GetLineSpan().StartLinePosition.Line == commentTrivia.GetLineSpan().EndLinePosition.Line && !nodeAfterTrivia.GetTrailingTrivia().Contains(commentTrivia))
```
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1120CodeFixProvider.cs]: The if and return statements below can be replaced by a single return statement.

File: StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1120CodeFixProvider.cs [hunk 3]
```diff
@@ -53,27 +53,23 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             var nodesToRemove = new List<SyntaxTrivia>();
```
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1120CodeFixProvider.cs]: `var node` above and `nodesToRemove` could be renamed to show that they are about trivia and not actual nodes.

**Comment by vweijsters** [comment]
:+1: Changes look good to me. I added a few minor remarks.

File: StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1120CodeFixProvider.cs (hunk 2 above)
Comment (Noryoko) [inline StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1120CodeFixProvider.cs]: `GetLine()` and GetEndLine() are only available for `SyntaxToken` it seems.

Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1120CodeFixProvider.cs]: `GetLine()` is indeed. I would prefer that you add the missing extension methods to `LocationHelpers`, but I leave the choice to you.

**Comment by Noryoko** [comment]
Added the requested extension methods.

**Pull request merged**